File: Data Structures & Algorithms/duplicate-integer/submission-1.js
class Solution {
   hasDuplicate(nums) {
    let freq={}
    for (let i of nums){
        if(freq[i]){
            freq[i]+=1;
             }
        else {freq[i]=1}
    }
    for(let key in freq){
        if(freq[key]>1){
            return true
        }

       }
       return false
   }
}
let s=new Solution;

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        product=[1 for _ in range(len(nums))]
        left=1
        right=1
        for i in range(0,len(product)):
            product[i]*=left
            left*=nums[i]
        for j in range(len(product)-1,-1,-1):
            product[j]*=right
            right*=nums[j]
        return product
        
        